api/control/v1alpha1/types.go

```diff
@@ -144,6 +135,37 @@ type DownsyncPolicyClause struct {
 	// `statusCollection` holds the rules for collecting status from the WECs.
 	// +optional
 	StatusCollection *StatusCollection `json:"statusCollection,omitempty"`
+
+	// WantSingletonReportedState, in short, indicates an expectation
+	// that the matching workload objects are distributed to exactly one WEC
+	// and requests that the `.status` of such objects propagate from the WEC
+	// to the WDS.
+	//
+	// For a precise description, start with a couple of definitions.
+	// For a given workload object, _singleton status return is requested_
+	// if and only if there exists at least one BindingPolicy or Binding
+	// that has `wantSingletonReportedState==true` in a clause that
+	// matches/references the workload object.
+	//
+	// The _qualified WEC set_ of a workload object is the set of WECs that are
+	// associated with that workload object by at least one BindingPolicy or Binding
+	// that has `wantSingletonReportedState==true` in a clause that
+	// matches/references the workload object.
+	//
+	// For a given workload object, while singleton status return is requested,
+	// KubeStellar maintains a label on the object whose name (key) is
+	// `kubestellar.io/executing-count` and whose value is a string representation
+	// of the size of the qualified WEC set of that object.
+	// While singleton status return is _not_ requested, KubeStellar suppresses
+	// the existence of a label with that name (key).
+	// While singleton status return is requested _and_ the size of the qualified
+	// WEC set is 1, KubeStellar propagates the object's `.status` from that WEC
+	// to the `.status` section of the object in the WDS.
+	// While either singleton status return is NOT requested or the size of the
+	// qualified WEC set is NOT 1, the `.status` of the object in the WDS does
+	// not have contents propagated there from a WEC by KubeStellar.
+	// +optional
+	WantSingletonReportedState bool `json:"wantSingletonReportedState,omitempty"`
```

Comment:
Why not move this inside `StatusCollection`?

Reply:
Because `StatusCollection` is just a set of references to `StatusCollector` objects. The singleton state return does not modulate the meaning of those or otherwise affect the `CombinedStatus` objects.

Something that I _could_ support would be to refactor `DownsyncPolicyClause` so that it has just two fields: `DownsyncObjectTest` and `DownsyncModulation` (or some other appropriate new name), where the latter puts together all the things that modulate the downsync behavior --- i.e., `CreateOnly bool`, `WantSingletonReportedState bool`, and `StatusCollectors []string`.

Reply:
That sounds good. My main concern is that such a flag should be at the test-level, since it may make no sense to apply it to some resources in a workload. Therefore the user may duplicate/narrow-down object selectors and enable status collection in only some of them.

Reply:
OK. I made a new PR in that style. Please see #2463 .